pkg/rc/replication_controller.go

```diff
@@ -277,14 +282,7 @@ func (rc *replicationController) addPods(current types.PodLocations, eligible []
 			txn, cancelFunc = rc.newAuditingTransaction(context.Background(), txn.Nodes())
 		}
 		if len(possibleSorted) < i+1 {
-			errMsg := fmt.Sprintf(
-				"Not enough nodes to meet desire: %d replicas desired, %d currentNodes, %d eligible. Scheduled on %d nodes instead.",
-				rc.ReplicasDesired, len(currentNodes), len(eligible), i,
-			)
-			err := rc.alerter.Alert(rc.alertInfo(errMsg))
-			if err != nil {
-				rc.logger.WithError(err).Errorln("Unable to send alert")
-			}
+			err := rc.allocateNodes(currentNodes, eligible)
```

Comment:
If the goal of this PR is to allow node transfers, I think the right place for this is actually within `checkForIneligible()` on line 237 (of course that function should be renamed). "ineligible" means that the RC is scheduled on a node that it shouldn't be scheduled on, it seems like as written the goal of this code is being able to scale an RC beyond what has been allocated. We shouldn't even need a node transfer for that case since it's simply adding more!

Reply:
Are you suggesting that we should not be allocating nodes here at all? I took it as there are two cases in which we allocate: 1) the ineligible case, and 2) the not enough nodes case where we simply add one without performing a node transfer.